students/brian_warn/session03/mailroom.py

```diff
@@ -0,0 +1,112 @@
+            Print the email to the terminal and return to the original prompt. 
+        '''
+
+        # Get the name
+        ty_name = None
+        while (True):
+            ty_name = str(input("Please enter the donor's name or type 'list' to see donor names. (Enter 'menu' to return to the original menu.) --> ")).strip()
+            if ty_name == 'list':
+                print("List of Donors:")
+                for ty_name in donor_gifts:
+                    print(ty_name)
+            elif ty_name == 'menu':
+                return
+            else:
+                break
+
+        # Enter the donation amount. Input taken as an integer-only value
+        while(True):
+            donation_amount = int(input("Please enter the donation amount (Enter 'menu' to return to the original menu.) --> "))
+            if donation_amount == 'menu':
+                return
+            else:
+                break
+
+        # Create an entry for a new name
+        if ty_name not in donor_gifts:
+            donor_gifts[ty_name] = []
+
+        # Store the donation and send a thank you
+        donor_gifts[ty_name].append(donation_amount)
+        # Generate thank you note.  Format placeholders don't have any padding in order to produce a clean message.
+        print("\nDear {d:}, \nThank you for your generous donation of ${t:<,.2f}.".format(d=ty_name, t=donation_amount))
+
+
+# create_report() -->
+
+    def create_report():
+        ''' Create the report of donors and amounts. Present in formatted output. '''
+        donor_report = []
+        # Perform calculations for the number of gifts, total amount given by a donor, and a donor's average gift.
+        for name,gifts in donor_gifts.items():
+            total = sum(gifts)
+            number_gifts = len(gifts)
+            average_gift = total / number_gifts
+            donor_report.append((name, total, number_gifts, average_gift))
+
+        # Print the Report
+        print("{: <30s} {: >19s}{: >14s}{: >15s}".format("Donor Name", "Amount Donated", "Total Gifts", "Average Gift"))
+        print("-"*30, "+ ", "-"*16, "+", "-"*11, "+", "-"*12)
+        for (name, total, number_gifts, average_gift) in donor_report:
+            print("{d: <30} ${t: 18.2f}{n: >14d}{a: >15.2f}".format(
+                d=name,
+                t=total,
+                n=number_gifts,
+                a=average_gift))
+
+# show_menu -->
+    def show_menu():
+        '''
+        description: Present a menu of the choices to add an item, show current items, or exit the program.
+        :rtype: string
+        :return: string type
+        '''
+        while (True):
+            print("""
+                Menu of Options
+                1) Send a Thank You
+                2) Create a Report
+                3) Quit
+                """)
+            strUserChoice = str(input("Which option would you like to perform? [1 to 3] - "))
```

Comment:
no need to call str() on input results -- it always returns a string.

ANd I'm not a fan of "hungarian notation" == of course it's a string, why not "user_choice" (no capitalization -- PEP8 style)